src/dbnode/namespace/schema.go

```diff
@@ -278,32 +297,80 @@ func marshalFileDescriptors(fdList []*desc.FileDescriptor) ([][]byte, error) {
 	return dlist, nil
 }
 
-func LoadSchemaRegistryFromFile(schemaReg SchemaRegistry, nsID ident.ID, protoFile string, msgName string, importPathPrefix ...string) error {
-	out, _ := parseProto(protoFile, importPathPrefix...)
+// AppendSchemaOptions appends to a provided SchemaOptions with a new version of schema.
+// The new version of schema is parsed out of the provided protoFile/msgName/contents.
+// schemaOpt: the SchemaOptions to be appended to, if nil, a new SchemaOption is created.
+// deployID: the version ID of the new schema.
+// protoFile: name of the top level proto file.
+// msgName: name of the top level proto message.
+// contents: map of name to proto strings.
+//          Except for the top level proto file, other imported proto files' key must be exactly the same
+//          as how they are imported in the import statement:
+//          E.g. if import.proto is imported as below
+//          import "mainpkg/imported.proto";
+//          Then the map key for improted.proto must be "mainpkg/imported.proto"
+//          See src/dbnode/namesapce/kvadmin test for example.
+func AppendSchemaOptions(schemaOpt *nsproto.SchemaOptions, protoFile, msgName string, contents map[string]string, deployID string) (*nsproto.SchemaOptions, error) {
```

Comment:
Maybe a quick check that deployID is not "" and also that it doesn't exist as any of the previous deployIDs (to prevent conflicts)?

Reply:
will do